webpack.config.js

```diff
@@ -65,12 +62,10 @@ module.exports = {
   devtool: 'source-map',
   plugins: [
     new webpack.DefinePlugin({
-      'COMPILE_CONSTANTS': JSON.stringify({
-        PAGE_TITLE,
-        DEFAULT_SITE,
-        ENABLE_GOTO_OTHER_SITE: false,
-        DEVELOPER_MODE,
-      })
+      'process.env.PTTCHROME_PAGE_TITLE': JSON.stringify(process.env.PTTCHROME_PAGE_TITLE || 'PttChrome'),
+      'process.env.DEFAULT_SITE': JSON.stringify(PRODUCTION_MODE ? 'wsstelnet://ws.ptt.cc/bbs' : 'wstelnet://localhost:8080/bbs'),
+      'process.env.ENABLE_GOTO_OTHER_SITE': JSON.stringify(false),
+      'process.env.DEVELOPER_MODE': JSON.stringify(DEVELOPER_MODE),
```

Comment:
Why process.env is preferred? I thought this is a nodejs-only thingy.

Reply:
`process.env` is used in several front-end project to mark as the reference point of compile time constants. For example, `React` checks `process.env.NODE_ENV === 'production'` to see if it should turning off debug messages.

I prefer defining compile time constants like `process.env.SOME_VAR` over defining it directly `SOME_VAR`, because it gives us better hint that it's a compile time constant (not some runtime `const SOME_VAR = 'a'` defined at the top of the module).

Reply:
React uses `process.env` for compile time constant:
https://github.com/facebook/react/blob/d9c1dbd61772f8f8ab0cdf389e70463d704c480b/packages/react/npm/index.js

Reply:
Um, ok